kpet/data.py

```diff
@@ -296,6 +299,10 @@ def __init__(self, data):
             ),
             data
         )
+        if self.sets is None:
+            self.sets = set()
```

Comment:
I think it's better to just keep `self.sets` None, if it's not set. This way we allow empty set specification and simplify logic of detecting if the sets were specified or not.

Reply:
The convert_sets function right now will change it into an empty set anyways, I suppose we want to change that as well then?